mumax3c/drivers/driver.py

```diff
@@ -37,246 +22,98 @@ def __init__(self, **kwargs):
-                # Update system's magnetisation. An example .omf filename:
-                # test_sample-Oxs_TimeDriver-Magnetization-01-0000008.omf
-                ovffiles = glob.iglob(os.path.join(f"{system.name}.out", "m_full*.ovf"))
-                lastovffile = sorted(ovffiles)[-1]
-                # pass Field.array instead of Field for better performance
-                # Mumax3 norm changes so need to set back to old norm
-                norm_field = system.m.norm
-                system.m.value = df.Field.fromfile(lastovffile).array
-                system.m.norm = norm_field
-
-                # Update system's datatable.
-                if isinstance(self, mc.TimeDriver):
-                    x = "t"
-                elif isinstance(self, mc.MinDriver):
-                    x = "t"  # TODO correct iteration
-                elif isinstance(self, mc.RelaxDriver):
-                    x = "t"  # TODO correct iteration
-                elif isinstance(self, mc.HysteresisDriver):
-                    x = "B_hysteresis"
-                system.table = ut.Table.fromfile(
-                    os.path.join(f"{system.name}.out", "table.txt"), x=x
-                )
-
-        if compute is None:
-            system.drive_number += 1
+    def _call(self, system, runner, verbose=1, dry_run=False, **kwargs):
+        if runner is None:
+            runner = mc.runner.runner
+        if dry_run:
+            return runner.call(argstr=self._mx3filename(system), dry_run=True)
         else:
-            system.compute_number += 1
-
-        # remove information about fixed cells for subsequent runs
-        # TODO
-
-    def _time_dependence(self, term, **kwargs):
-        try:
-            tmax = kwargs["t"]
-        except KeyError:
-            msg = (
-                f"Time-dependent term {term.__class__.__name__=} must be "
-                "used with time driver."
-            )
-            raise RuntimeError(msg)
-        ts = np.arange(0, tmax + term.dt, term.dt)
-        try:  # vector output from term.func
-            tlist = [list(term.func(t)) for t in ts]
-            dtlist = (np.gradient(tlist)[0] / term.dt).tolist()
-        except TypeError:  # scalar output from term.func
-            tlist = [term.func(t) for t in ts]
-            dtlist = list(np.gradient(tlist) / term.dt)
-        term.tlist = tlist
-        term.dtlist = dtlist
+            runner.call(argstr=self._mx3filename(system), verbose=verbose)
+
+    def _read_data(self, system):
+        # Update system's magnetisation. An example .ovf filename: m_full000000.ovf
+        ovffiles = pathlib.Path(f"{system.name}.out").glob("m_full*.ovf")
+        lastovffile = sorted(ovffiles)[-1]
+        # pass Field.array instead of Field for better performance
+        # Mumax3 norm changes so need to set back to old norm
+        norm_field = system.m.norm
+        system.m.value = df.Field.fromfile(str(lastovffile)).array
+        system.m.norm = norm_field
```

Comment:
@lang-m This line raises a `ValueError` from `discretisedfield` with message "Cannot normalise field with zero values." For a better context, I was trying to run the skyrmion in a disk example with `mumax3c`. I am guessing the `Ms = 0` values are creating the problem? Why do we need this check in `discretisedfield`?

PS: For the time I have circumvent this by `system.m = df.Field.fromfile(str(lastovffile))`.